File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.js
class Solution {
    /**
     * @param {string} s
     * @return {number}
     */
    lengthOfLongestSubstring(s) {
        if(s.length === 0){
            return 0;
        }
        let left = 0;
        let right = 0;
        let max = -0;
        const set = new Set();
        while(right<s.length){
            if(!set.has(s[right])){
                set.add(s[right]);
                max = Math.max(max,right-left+1);
                right++;
            }
            else{
                set.delete(s[left]);
                left++;
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-1.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int left = 1 , right = 1;
        for(int pile : piles){
            right = Math.max(right,pile);
        }
        while(left<right){
            int mid = left+(right-left)/2;
            if(canFinish(mid,piles,h)){
                right = mid;
            }
            else{
                left= mid+1;
            }
        }
        return left;
    }
    private boolean canFinish(int mid, int[] piles, int h){
        int hours = 0;
        for(int pile : piles){
            hours += Math.ceil((double)pile/mid);
        }
        return h>=hours;
    }
}
